arduino/sketch_read_transmit_from_sd/sketch_read_transmit_from_sd.ino

```diff
@@ -27,28 +27,28 @@ File openTheFileFromSDCard(String fileName) {
     Serial.println("error opening: " + fileName);
     return file;
   }
-
-  Serial.println(fileName);
   return file;
 }
 
 /*
   Reads the file passed in as an argument from the SD card and transmit it over the serial output channel line by line with '\n' as the line delimitter. Skips null lines.
 */
-void readFromSDCardToSerialOutputLineByLine(File file) {   // TODO: delay value can be passed in as an argument
+void readFromSDCardToSerialOutput(File file) {
   // read from the file until there's nothing else in it
-  String line;
+  delay(1000);
   // we could use something here like line.reserve(numberOfBytes) to avoid heap memory fragmentation
-  while (file.available()) {
-    line = file.readStringUntil('\n');
-    line.trim();
-    if (line != "") {
-      Serial.println(line);
-      // TODO: Delay can be added here to transmit data over the serial channel for a specified constant rate.
-    }  // Skip blank null lines.
-  }
+if(file){
+// Reading the whole file
+    while (file.available()) {
+      Serial.write(file.read());
+   }
+   Serial.write("Recorded");
```

Comment:
> Why are we using a letter R to detect the EOF? Are we adding a R to the end of each set?

@CloudyPadmal Kind of yes but not to the dataset.

Reply:
Now only I noticed this. Are we writing the whole file in a single write??

Reply:
Yes. But should i print it line by line?

Reply:
Of course. Revert back to the previous implementation.

Reply:
Okay.